File: app/src/main/java/com/productivity/habits/data/scheduler/NoOpHabitReminderScheduler.kt
package com.productivity.habits.data.scheduler

import com.productivity.habits.data.local.entity.HabitEntity
import com.productivity.habits.domain.scheduler.HabitReminderScheduler
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class NoOpHabitReminderScheduler @Inject constructor() : HabitReminderScheduler {
    override suspend fun schedule(habit: HabitEntity) {
        // No-op for Phases 1-4. Wired to AlarmManager & WorkManager in Phase 5.
    }

    override suspend fun cancel(habitId: String) {
        // No-op for Phases 1-4. Wired to AlarmManager & WorkManager in Phase 5.
    }

    override suspend fun rescheduleAll() {
        // No-op for Phases 1-4. Wired to AlarmManager & WorkManager in Phase 5.
    }
}

File: lib/data/schedulers/no_op_habit_reminder_scheduler.dart
import '../../domain/models/habit.dart';
import '../../domain/schedulers/habit_reminder_scheduler.dart';

class NoOpHabitReminderScheduler implements HabitReminderScheduler {
  const NoOpHabitReminderScheduler();

  @override
  Future<void> schedule(Habit habit, {bool catchUpIfDue = false}) async {
    // No-op for Phases 1-4. Wired to Local Notifications & WorkManager in Phase 5.
  }

  @override
  Future<void> cancel(String habitId) async {
    // No-op for Phases 1-4. Wired to Local Notifications & WorkManager in Phase 5.
  }

  @override
  Future<void> rescheduleAll() async {
    // No-op for Phases 1-4. Wired to Local Notifications & WorkManager in Phase 5.
  }
}
